spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkWriteConf.java

```diff
@@ -214,6 +214,10 @@ private boolean fanoutWriterEnabled(boolean defaultValue) {
   }
 
   public FileFormat deleteFileFormat() {
+    if (!(table instanceof BaseMetadataTable) && TableUtil.formatVersion(table) >= 3) {
```

Comment:
See https://github.com/apache/iceberg/pull/11986#discussion_r1919554870 for why I changed this